botocore/stub.py

```diff
@@ -362,3 +388,13 @@ def _validate_response(self, operation_name, service_response):
             raise ParamValidationError(
                 report=(
                     "Service response should only contain ResponseMetadata."))
+
+
+class StubRequest(object):
+    """
+    A tracked request that was made to a stubbed client. Has two attributes,
+    ``operation_name`` and ``params``.
+    """
+    def __init__(self, operation_name, params):
+        self.operation_name = operation_name
```

Comment:
Maybe I should say "read-only". People can of course still access those attributes, but it's generally understood that they're internal and should not be used. You could make it more forceful with `__operation_name`, but there's no need to go so far.